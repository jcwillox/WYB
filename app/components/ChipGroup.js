import React, { useState } from "react";
import { FlatList } from "react-native";
import { Chip } from "react-native-paper";

function ChipGroup({ items, onSelectedChange }) {
  const [selected, setSelected] = useState(new Set());
  const handleSelected = (item) => {
    let updatedSet = new Set(selected);
    if (updatedSet.has(item)) updatedSet.delete(item);
    else updatedSet.add(item);

    setSelected(updatedSet);

    // future to ensure we update the selected item, before processing change
    if (onSelectedChange)
      setTimeout(() => {
        onSelectedChange(updatedSet);
      }, 0);
  };

  const renderItem = ({ item }) => (
    <Chip
      mode="outlined"
      style={{ marginLeft: 8 }}
      onPress={() => handleSelected(item)}
      selected={selected.has(item)}
    >
      {item}
    </Chip>
  );
  return (
    <FlatList
      data={items}
      renderItem={renderItem}
      keyExtractor={(item) => item}
      horizontal
      extraData={selected}
    />
  );
}

export default ChipGroup;
