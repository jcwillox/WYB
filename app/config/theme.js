import { Colors, DefaultTheme } from "react-native-paper";

const LightTheme = {
  ...DefaultTheme,
  roundness: 8,
  colors: {
    ...DefaultTheme.colors,
    primary: Colors.indigoA700,
    accent: Colors.pinkA700,
    stars: Colors.amber500,
    background: Colors.grey50,
    trash: Colors.red700,
    lightGrey: Colors.grey100,
    mediumGrey: Colors.grey400,
  },
};

export default LightTheme;
