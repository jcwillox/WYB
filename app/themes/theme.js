import { DarkTheme, DefaultTheme } from "react-native-paper";
import deepmerge from "deepmerge";

const LightTheme = deepmerge(DefaultTheme, {
  // roundness: 8,
  colors: {
    primary: "#304ffe", // Indigo A700
    listing: "#00bfa5", // Teal A700
    account: "#c51162", // Pink A700
    stars: "#ffc107", // Amber 500
  },
  // animation: {
  //   scale: 50,
  // },
});

export default LightTheme;

// {
//   primary: "",
//   accent: "#304ffe",
//   background: "#F2F2F2",
//   backgroundDark: "#757575",
// };
