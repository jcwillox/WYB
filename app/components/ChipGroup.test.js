import React from "react";
import renderer from "react-test-renderer";
import { Colors, Provider as PaperProvider } from "react-native-paper";
import ChipGroup from "./ChipGroup";
import theme from "../config/theme";

it("ensure <ChipGroup /> renders correctly", () => {
  let selected = new Set(["Shops"]);
  let categories = [
    { name: "Attractions", color: Colors.deepPurple500 },
    { name: "Shops", color: Colors.amber800 },
    { name: "Restaurants", color: Colors.green600 },
    { name: "Hotels", color: Colors.cyan600 },
  ];

  const tree = renderer
    .create(
      <PaperProvider theme={theme}>
        <ChipGroup items={categories} selected={selected} />
      </PaperProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
