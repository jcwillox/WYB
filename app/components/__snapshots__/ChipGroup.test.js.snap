// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensure <ChipGroup /> renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "color": "#673ab7",
          "name": "Attractions",
        },
        Object {
          "color": "#ff8f00",
          "name": "Shops",
        },
        Object {
          "color": "#43a047",
          "name": "Restaurants",
        },
        Object {
          "color": "#00acc1",
          "name": "Hotels",
        },
      ]
    }
    disableVirtualization={false}
    extraData={
      Set {
        "Shops",
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(103, 58, 183, 0.29)",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "marginRight": 8,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingLeft": 4,
                    "position": "relative",
                  },
                  Object {
                    "paddingRight": 4,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "marginVertical": 4,
                        "minHeight": 24,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "color": "rgba(103, 58, 183, 0.87)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Attractions
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(235, 235, 235)",
              "borderColor": "rgba(255, 143, 0, 0.29)",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "marginRight": 8,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
                "selected": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingLeft": 4,
                    "position": "relative",
                  },
                  Object {
                    "paddingRight": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "padding": 4,
                    },
                    null,
                  ]
                }
              >
                <Text />
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "marginVertical": 4,
                        "minHeight": 24,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "color": "rgba(255, 143, 0, 0.87)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "marginLeft": 4,
                        "marginRight": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Shops
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(67, 160, 71, 0.29)",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "marginRight": 8,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingLeft": 4,
                    "position": "relative",
                  },
                  Object {
                    "paddingRight": 4,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "marginVertical": 4,
                        "minHeight": 24,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "color": "rgba(67, 160, 71, 0.87)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Restaurants
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(0, 172, 193, 0.29)",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingLeft": 4,
                    "position": "relative",
                  },
                  Object {
                    "paddingRight": 4,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "marginVertical": 4,
                        "minHeight": 24,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "color": "rgba(0, 172, 193, 0.87)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Hotels
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
